### DISCUSSION TITLE
Cant get it to work for some reason. (noob here)

### DISCUSSION THREAD

**Discussion by TibiaBubble** [cSharpBastard/mmm-nordic-electrical-spot-prices#2] [Q&A]
I have been trying to put the modul in the config file and been trying to play around a bit cant get it to work.

if I remove the text it boots just fine.

```

let config = {

......

        modules: [
                {
                        module: "alert",
                },
                {
                        module: "updatenotification",
                        position: "top_bar"
                },
                {
                        module: "clock",
                        position: "top_left"
                },



                {
                        module: "mmm-nordic-electrical-spot-prices",
                        position: "top_right"
                },
........

```

**Comment by cSharpBastard** [comment]
Do you mean that the MagicMirror doesn't start with the module configured?
Do you have the module under modules directly? Like MagicMirror/modules/mmm-nordic-electrical-spot-proces
Check that the user running the module has access to that path.

**Comment by TibiaBubble** [comment]
the permissions are set to 775 (with chmod), still not working :(

the modul is pulled with the git clone command from the page in to the modules page (picture here: https://ibb.co/1JmYNq1 ).


> dietpi@DietPi:~/MagicMirror$ npm run start
> 
> > magicmirror@2.22.0 start
> > DISPLAY="${DISPLAY:=:0}" ./node_modules/.bin/electron js/electron.js
> 
> [19.01.2023 08:34.41.767] [LOG]   Starting MagicMirror: v2.22.0
> [19.01.2023 08:34.41.785] [LOG]   Loading config ...
> [19.01.2023 08:34.41.822] [LOG]   Loading module helpers ...
> [19.01.2023 08:34.41.840] [LOG]   No helper found for module: alert.
> [19.01.2023 08:34.41.912] [LOG]   Initializing new module helper ...
> [19.01.2023 08:34.41.915] [LOG]   Module helper loaded: updatenotification
> [19.01.2023 08:34.41.919] [LOG]   No helper found for module: clock.
> [19.01.2023 08:34.41.971] [ERROR] WARNING! Could not load config file. Starting with default configuration. Error found: Error: Cannot find module 'request'
> Require stack:
> - /home/dietpi/MagicMirror/modules/mmm-nordic-electrical-spot-prices/node_helper.js
> - /home/dietpi/MagicMirror/js/app.js
> - /home/dietpi/MagicMirror/js/electron.js
> - /home/dietpi/MagicMirror/node_modules/electron/dist/resources/default_app.asar/main.js
> - 
> [19.01.2023 08:34.41.973] [LOG]   Loading module helpers ...
> [19.01.2023 08:34.41.978] [LOG]   No helper found for module: alert.
> [19.01.2023 08:34.41.982] [LOG]   Initializing new module helper ...
> [19.01.2023 08:34.41.986] [LOG]   Module helper loaded: updatenotification
> [19.01.2023 08:34.41.990] [LOG]   No helper found for module: clock.
> [19.01.2023 08:34.41.997] [ERROR] App threw an error during load
> [19.01.2023 08:34.42.002] [ERROR] Error: Cannot find module 'request'
> Require stack:
> - /home/dietpi/MagicMirror/modules/mmm-nordic-electrical-spot-prices/node_helper.js
> - /home/dietpi/MagicMirror/js/app.js
> - /home/dietpi/MagicMirror/js/electron.js
> - /home/dietpi/MagicMirror/node_modules/electron/dist/resources/default_app.asar/main.js
> - 
>     at Module._resolveFilename (node:internal/modules/cjs/loader:963:15)
>     at n._resolveFilename (node:electron/js2c/browser_init:2:109416)
>     at Module._resolveFilename (/home/dietpi/MagicMirror/node_modules/module-alias/index.js:49:29)
>     at Module._load (node:internal/modules/cjs/loader:811:27)
>     at f._load (node:electron/js2c/asar_bundle:2:13328)
>     at Module.require (node:internal/modules/cjs/loader:1035:19)
>     at require (node:internal/modules/cjs/helpers:102:18)
>     at Object.<anonymous> (/home/dietpi/MagicMirror/modules/mmm-nordic-electrical-spot-prices/node_helper.js:3:17)
>     at Module._compile (node:internal/modules/cjs/loader:1141:14)
>     at Module._extensions..js (node:internal/modules/cjs/loader:1196:10)
> [19.01.2023 08:34.42.007] [ERROR] Whoops! There was an uncaught exception...
> [19.01.2023 08:34.42.024] [ERROR] Error: Cannot find module 'request'
> Require stack:
> - /home/dietpi/MagicMirror/modules/mmm-nordic-electrical-spot-prices/node_helper.js
> - /home/dietpi/MagicMirror/js/app.js
> - /home/dietpi/MagicMirror/js/electron.js
> - /home/dietpi/MagicMirror/node_modules/electron/dist/resources/default_app.asar/main.js
> - 
>     at Module._resolveFilename (node:internal/modules/cjs/loader:963:15)
>     at n._resolveFilename (node:electron/js2c/browser_init:2:109416)
>     at Module._resolveFilename (/home/dietpi/MagicMirror/node_modules/module-alias/index.js:49:29)
>     at Module._load (node:internal/modules/cjs/loader:811:27)
>     at f._load (node:electron/js2c/asar_bundle:2:13328)
>     at Module.require (node:internal/modules/cjs/loader:1035:19)
>     at require (node:internal/modules/cjs/helpers:102:18)
>     at Object.<anonymous> (/home/dietpi/MagicMirror/modules/mmm-nordic-electrical-spot-prices/node_helper.js:3:17)
>     at Module._compile (node:internal/modules/cjs/loader:1141:14)
>     at Module._extensions..js (node:internal/modules/cjs/loader:1196:10) {
>   code: 'MODULE_NOT_FOUND',
>   requireStack: [
>     '/home/dietpi/MagicMirror/modules/mmm-nordic-electrical-spot-prices/node_helper.js',
>     '/home/dietpi/MagicMirror/js/app.js',
>     '/home/dietpi/MagicMirror/js/electron.js',
>     '/home/dietpi/MagicMirror/node_modules/electron/dist/resources/default_app.asar/main.js',
>     undefined
>   ]
> }
> [19.01.2023 08:34.42.025] [ERROR] MagicMirror² will not quit, but it might be a good idea to check why this happened. Maybe no internet connection?
> [19.01.2023 08:34.42.026] [ERROR] If you think this really is an issue, please open an issue on GitHub: https://github.com/MichMich/MagicMirror/issues
> [19.01.2023 08:34.44.621] [LOG]   Launching application.
> MESA-LOADER: failed to retrieve device information
> MESA-LOADER: failed to retrieve device information
> MESA-LOADER: failed to retrieve device information
> ../../buildtools/third_party/libc++/trunk/include/vector:1425: assertion __n < size() failed: vector[] index out of bounds[1199:0119/083507.996963:ERROR:gpu_process_host.cc(991)] GPU process exited unexpectedly: exit_code=6
> MESA-LOADER: failed to retrieve device information
> MESA-LOADER: failed to retrieve device information
> MESA-LOADER: failed to retrieve device information
> ../../buildtools/third_party/libc++/trunk/include/vector:1425: assertion __n < size() failed: vector[] index out of bounds[1199:0119/083516.022993:ERROR:gpu_process_host.cc(991)] GPU process exited unexpectedly: exit_code=6
> MESA-LOADER: failed to retrieve device information
> MESA-LOADER: failed to retrieve device information
> MESA-LOADER: failed to retrieve device information
> ../../buildtools/third_party/libc++/trunk/include/vector:1425: assertion __n < size() failed: vector[] index out of bounds[1199:0119/083523.288253:ERROR:gpu_process_host.cc(991)] GPU process exited unexpectedly: exit_code=6
> [19.01.2023 08:42.30.003] [LOG]   Shutting down server...
> [19.01.2023 08:42.30.117] [LOG]   Stopping module helper: updatenotification
> [19.01.2023 08:42.30.118] [LOG]   Stopping module helper: updatenotification
> [19.01.2023 08:42.30.158] [ERROR] Whoops! There was an uncaught exception...
> [19.01.2023 08:42.30.225] [ERROR] TypeError: Cannot read properties of undefined (reading 'close')
>     at App.stop (/home/dietpi/MagicMirror/js/app.js:274:14)
>     at App.<anonymous> (/home/dietpi/MagicMirror/js/electron.js:159:7)
>     at App.emit (node:events:513:28)
>     at appMethod (node:electron/js2c/browser_init:2:24153)
>     at Object.execute (node:electron/js2c/browser_init:2:28291)
>     at MenuItem.click (node:electron/js2c/browser_init:2:30120)
>     at a._executeCommand (node:electron/js2c/browser_init:2:35562)
> [19.01.2023 08:42.30.232] [ERROR] MagicMirror² will not quit, but it might be a good idea to check why this happened. Maybe no internet connection?
> [19.01.2023 08:42.30.234] [ERROR] If you think this really is an issue, please open an issue on GitHub: https://github.com/MichMich/MagicMirror/issues
> 
> 
> 
>

**Reply by TibiaBubble** [reply]
I can add that if I remove the text in the config file, it all starts fine and I get no errors :(

**Reply by cSharpBastard** [reply]
I'll look into it. In the mean time try to install the Python module requests.
`python3 -m pip install requests`

**Reply by TibiaBubble** [reply]
Didn’t work 😢

**Comment by cSharpBastard** [comment]
You probably missed the dependencies (electron, moment and request). The error message says missing request.
cd into you MagicMirror folder and run:
`npm install request`

**Reply by TibiaBubble** [reply]
This command kinda killed my dietpi 😂😂😂

installing raspberry OS now 🤣😂

**Reply by cSharpBastard** [reply]
Sorry to hear that!
You're taking it well though! 😀

**Reply by TibiaBubble** [reply]
Maybe the issue was that I was running it on dietpi on a raspberry pi zero 2 w 😅

Can’t get it to work now ether, thinking maybe my memory card is broken 😂
